torch/_dynamo/output_graph.py

```diff
@@ -1441,10 +1445,22 @@ def remove_unused(node):
                     arg = node.meta["grapharg"]
                     if isinstance(arg, BackwardStateGraphArg):
                         continue
+                    if isinstance(node.meta["grapharg"].example, torch.ScriptObject):
```

Comment:
animesh: check whether methods are called of script object and decide whether to delete it or not.

Reply:
It turns out the current implementation might be correct: used_symbols is not used to prune the script object (it's pruned in 1439) but used to prune SymInt intputs. In that case , we should add the symints in the contents of torch.ScriptObject to used_symbols to be safe.